leshan-core-cf/src/main/java/org/eclipse/leshan/core/californium/EndpointContextUtil.java

```diff
@@ -43,10 +42,11 @@ public static Identity extractIdentity(EndpointContext context) {
             } else if (senderIdentity instanceof RawPublicKeyIdentity) {
                 PublicKey publicKey = ((RawPublicKeyIdentity) senderIdentity).getKey();
                 return Identity.rpk(peerAddress, publicKey);
-            } else if (senderIdentity instanceof X500Principal || senderIdentity instanceof X509CertPath) {
-                // Extract common name
-                String x509CommonName = extractCN(senderIdentity.getName());
+            } else if (senderIdentity instanceof X500Principal) {
+                String x509CommonName = X509Util.extractCN(senderIdentity.getName());
                 return Identity.x509(peerAddress, x509CommonName);
+            } else if (senderIdentity instanceof X509CertPath) {
+                return Identity.x509(peerAddress, ((X509CertPath) senderIdentity).getPath());
```

Comment:
isn't useful to have the common name and the certificate chain when possible?

Reply:
The common name is extracted from the first certificate of the chain in the Identity class.